src/python/grpcio/grpc/aio/_interceptor.py

```diff
@@ -599,16 +599,14 @@ async def _invoke(
         """Run the RPC call wrapped in interceptors"""
 
         async def _run_interceptor(
-                interceptors: Iterator[UnaryUnaryClientInterceptor],
+                interceptors: List[UnaryUnaryClientInterceptor],
                 client_call_details: ClientCallDetails,
                 request: RequestType) -> _base_call.UnaryUnaryCall:
 
-            interceptor = next(interceptors, None)
-
-            if interceptor:
-                continuation = functools.partial(_run_interceptor, interceptors)
-
-                call_or_response = await interceptor.intercept_unary_unary(
+            if len(interceptors) > 0:
```

Comment:
Nit: According to the style guide (which I know we don't technically follow), this should be `if interceptors:`